service/src/initialized_service.rs

```diff
@@ -47,8 +51,48 @@ pub async fn start_is_initialized_server(port: u16) -> ServiceResult<()> {
 	Ok(())
 }
 
-/// Set initialized handler value to true.
-pub fn set_initialized() {
-	let mut initialized_lock = INITIALIZED_HANDLE.write();
-	*initialized_lock = true;
+/// Trait to query of a worker is considered fully initialized.
+pub trait IsInitialized {
+	fn is_initialized(&self) -> bool;
+}
+
+/// Tracker for initialization. Used by components that ensure these steps were taken.
+pub trait TrackInitialization {
+	fn registered_on_parentchain(&self);
+
+	fn sidechain_block_produced(&self);
+
+	fn worker_for_shard_registered(&self);
+}
```

Comment:
I had to extend the `is_initialized` service. We now have several conditions that lead us to report an `is_initialized` state. In particular, this includes:
* a validateer has produced a sidechain block
* the chain state `WorkerForShard` is not empty

This was also requested by #731

Reply:
I think this is very useful 👍